contents/swift/welcome-to-swift.md

```diff
@@ -12,44 +12,46 @@
+1. **iOS developers have good employment.** Mobile development becomes popular recently. A [survey in StackOverFlow](https://insights.stackoverflow.com/survey/2017#developer-profile-developer-type) shows that mobile development, although is significantly less than web development, still occupies large marketing along with desktop application development. In addition, iOS developer is well-paid. The [salary statistics](https://www.indeed.com/salaries/iOS-Developer-Salaries) shows that iOS developer in US can earn more than $120K per year. Therefore, learning Swift is helpful for your career.
 
-If you're just looking to take a dip into Swift, here are the [basics](https://guides.codepath.com/ios/Swift-Basics) you'll need to understand.
+1. **Swift is easy to learn.** Swift is aimed to be designed as a simple and understandable language by Apple. Its syntax resembles English, and thus is easy to read and understand. Any programmer being familiar with Python, Java or C++ can learn Swift qucikly. Even starting learning programming with swift is easy.
 
-And if you are really pressed for time, here's a couple of cheatsheets with code examples:
+1. **Swift is an [open source project](https://github.com/apple/swift).** Learning about Swift and contributing to the language may be a useful experience for those interested. In addition, its source code may be useful reading for those interested in learning good practices, or simply to find out more about how the language was implemented.
 
-- [Swift Cheat Sheet](https://mhm5000.gitbooks.io/swift-cheat-sheet/content/index.html) (web-friendly)
-- [Swift 3.0 Cheat Sheet and Quick Reference](https://koenig-media.raywenderlich.com/uploads/2014/06/RW-Swift-Cheatsheet-0_7.pdf) (print-friendly)
+Also note that Swift is not perfect. It is still very young, and therefore the language itself may have issues to be addressed. This may suffer the programmers during development.
+In addition, Swift still has limited number of libraries and tools, and many of the  resources dedicated to earlier Swift versions could be useless with the newer releases.
 
-# Cool Features
 
 Here are some cool features that you might not be familiar with but would definitely be useful to you if you are new to Swift. 
 
-## Type Inference
+### Benefit: Declare Variables Easily
 
-Swift supports type inference whereby the compiler automatically deduces the type of a variable during compilation by examining the values assigned to it.
+Swift supports type inference whereby the compiler automatically deduces the type of a variable during compilation by examining the values assigned to it. 
 
 ```swift
 var str1: String = "foo"
 var str2 = "foo" // compiler infers that str2 is of type String
 ```
 
-Do note that variables in Swift are statically typed.
+Unlike **Python** or **JavaScript** which use data dynamic type, variables in Swift are statically typed. Therefore, it will not lag the running time too much, and debugging will be easy.
```

Comment:
Better way to phrase the second sentence:

> Statically-typed languages have will have the code checked at compile-time instead of run-time, which eliminates many (often) trivial bugs early, which in turn makes debugging the program easier.